libpod/oci_conmon_common.go

```diff
@@ -370,15 +372,7 @@ func (r *ConmonOCIRuntime) KillContainer(ctr *Container, signal uint, all bool)
 		args = append(args, "kill", ctr.ID(), fmt.Sprintf("%d", signal))
 	}
 	if err := utils.ExecCmdWithStdStreams(os.Stdin, os.Stdout, os.Stderr, env, r.path, args...); err != nil {
-		// Update container state - there's a chance we failed because
-		// the container exited in the meantime.
-		if err2 := r.UpdateContainerStatus(ctr); err2 != nil {
-			logrus.Infof("Error updating status for container %s: %v", ctr.ID(), err2)
-		}
-		if ctr.ensureState(define.ContainerStateStopped, define.ContainerStateExited) {
-			return fmt.Errorf("%w: %s", define.ErrCtrStateInvalid, ctr.state.State)
```

Comment:
This is also going to break SigProxy. We need this to remain ErrCtrStateInvalid.

Reply:
Can you elaborate a bit more?  At least for kill and stop, the container is not locked, so we should not fiddle with the state.

Maybe we need to make this conditional?

Reply:
I can see only one place where KillContainer is run unlocked. I'll fix that.

We should not be touching KillContainer as such.

Reply:
You're right. It's unlocked during stop but not kill.

Reply:
https://github.com/containers/podman/pull/16323 catches the last case where KillContainer is run unlocked

Reply:
> I can see only one place where KillContainer is run unlocked. I'll fix that.

So you will fix #16142?

Reply:
Sure